src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -302,6 +315,15 @@ private void checkEnumDef(DetailAST ast) {
         }
     }
 
+    /**
+     * Check whether record has proper modifiers.
+     *
+     * @param recordDefAst ast node of type {@link TokenTypes#RECORD_DEF}
+     */
+    private void checkRecordDef(DetailAST recordDefAst) {
+        checkForRedundantModifier(recordDefAst, TokenTypes.LITERAL_STATIC);
```

Comment:
I think it can be inlined, no need to extract separate method

Reply:
Done

Reply:
no, lets leave it as it is, to avoid extending diff with unrelated changes